### DISCUSSION TITLE
Flow: Condition for checking if key exists

### DISCUSSION THREAD

**Discussion by lukas-schaetzle** [directus/directus#15479] [Archive]
I want to use the condition operation in a flow to check whether the paylod contains a key. My filter looks like this:

```
{
    "$trigger": {
        "payload": {
            "nav_menu": {
                "_nempty": true
             }
         }
    }
}
```

However, even when the trigger payload does not contain the `nav_menu` key the condition operation does not reject. Same behaviour if I use `_nnull` instead.

Not sure if this is a bug or I'm doing something wrong.

**Comment by br41nslug** [comment]
It sounds like you're running into this open issue 😬 https://github.com/directus/directus/issues/14571

**Comment by izoukhai** [comment]
You can create a "Run Script" operation and check the field this way:

`module.exports = async function({$trigger}) { if (!$trigger.payload.status) throw new Error("Status is not changed") else return {}; }`
